addons/ansible-service-broker/templates/deploy-ansible-service-broker.template.yaml

```diff
@@ -142,42 +170,81 @@ objects:
               scheme: HTTPS
             initialDelaySeconds: 15
             timeoutSeconds: 1
+        volumes:
+          - name: config-volume
+            configMap:
+              name: broker-config
+              items:
+              - key: broker-config
+                path: config.yaml
+          - name: asb-tls
+            secret:
+              secretName: asb-tls
+          - name: asb-auth-volume
+            secret:
+              secretName: asb-auth-secret
+          - name: asb-etcd-auth
+            secret:
+              secretName: broker-etcd-auth-secret
 
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: asb-etcd
+    labels:
+      app: etcd
+      service: asb-etcd
+  spec:
+    replicas: 1
+    selector:
+      app: etcd
+    strategy:
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          app: etcd
+          service: asb-etcd
+      spec:
+        serviceAccount: asb
+        containers:
         - image: ${ETCD_IMAGE}
           name: etcd
+          volumeMounts:
+            - name: etcd
+              mountPath: /data
+            - name: etcd-tls
+              mountPath: /etc/tls/private
+            - name: etcd-auth
+              mountPath: /var/run/etcd-auth-secret
           imagePullPolicy: IfNotPresent
           terminationMessagePath: /tmp/termination-log
           workingDir: /etcd
           args:
             - ${ETCD_PATH}
             - --data-dir=/data
-            - --listen-client-urls=http://0.0.0.0:2379
-            - --advertise-client-urls=http://0.0.0.0:2379
+            - --listen-client-urls=https://0.0.0.0:2379
+            - --advertise-client-urls=https://0.0.0.0:2379
```

Comment:
It will be better we set up the `--advertise-client-urls=https://asb-etcd.openshift-ansible-service-broker.svc:2379` here so that user can access the etcd by `etcdctl`. For details, please refer to the bug: https://bugzilla.redhat.com/show_bug.cgi?id=1514417